nbgrader/server_extensions/formgrader/apihandlers.py

```diff
@@ -90,10 +99,180 @@ def post(self, submission_id):
+    @web.authenticated
+    @check_xsrf
+    def get(self, student_id):
+        submissions = self.api.get_student_submissions(student_id)
+        self.write(json.dumps(submissions))
+
+
+class StudentNotebookSubmissionCollectionHandler(BaseApiHandler):
+    @web.authenticated
+    @check_xsrf
+    def get(self, student_id, assignment_id):
+        submissions = self.api.get_student_notebook_submissions(student_id, assignment_id)
+        self.write(json.dumps(submissions))
+
+
+class AssignHandler(BaseApiHandler):
+    @web.authenticated
+    @check_xsrf
+    def post(self, assignment_id):
+        self.write(json.dumps(self.api.assign(assignment_id)))
+
+
+class UnReleaseHandler(BaseApiHandler):
+    @web.authenticated
+    @check_xsrf
+    def post(self, assignment_id):
+        self.write(json.dumps(self.api.unrelease(assignment_id)))
+
+
+class ReleaseHandler(BaseApiHandler):
+    @web.authenticated
+    @check_xsrf
+    def post(self, assignment_id):
+        self.write(json.dumps(self.api.release(assignment_id)))
+
+
+class CollectHandler(BaseApiHandler):
+    @web.authenticated
+    @check_xsrf
+    def post(self, assignment_id):
+        self.write(json.dumps(self.api.collect(assignment_id)))
+
+
+class AutogradeHandler(BaseApiHandler):
+    @web.authenticated
+    @check_xsrf
+    def post(self, assignment_id, student_id):
+        self.write(json.dumps(self.api.autograde(assignment_id, student_id)))
+
+
 default_handlers = [
+    (r"/formgrader/api/assignments", AssignmentCollectionHandler),
+    (r"/formgrader/api/assignment/([^/]+)", AssignmentHandler),
+    (r"/formgrader/api/assignment/([^/]+)/assign", AssignHandler),
+    (r"/formgrader/api/assignment/([^/]+)/unrelease", UnReleaseHandler),
+    (r"/formgrader/api/assignment/([^/]+)/release", ReleaseHandler),
+    (r"/formgrader/api/assignment/([^/]+)/collect", CollectHandler),
+
+    (r"/formgrader/api/notebooks/([^/]+)", NotebookCollectionHandler),
+
+    (r"/formgrader/api/submissions/([^/]+)", SubmissionCollectionHandler),
+    (r"/formgrader/api/submission/([^/]+)/([^/]+)", SubmissionHandler),
+    (r"/formgrader/api/submission/([^/]+)/([^/]+)/autograde", AutogradeHandler),
+
+    (r"/formgrader/api/submitted_notebooks/([^/]+)/([^/]+)", SubmittedNotebookCollectionHandler),
+    (r"/formgrader/api/submitted_notebook/([^/]+)/flag", FlagSubmissionHandler),
+
     (r"/formgrader/api/grades", GradeCollectionHandler),
-    (r"/formgrader/api/comments", CommentCollectionHandler),
     (r"/formgrader/api/grade/([^/]+)", GradeHandler),
+
+    (r"/formgrader/api/comments", CommentCollectionHandler),
     (r"/formgrader/api/comment/([^/]+)", CommentHandler),
-    (r"/formgrader/api/submission/([^/]+)/flag", FlagSubmissionHandler)
+
+    (r"/formgrader/api/students", StudentCollectionHandler),
+    (r"/formgrader/api/student/([^/]+)", StudentHandler),
+
+    (r"/formgrader/api/student_submissions/([^/]+)", StudentSubmissionCollectionHandler),
```

Comment:
In general, all the handlers follow the convention of `/formgrader/api/plural_noun` for collections and `/formgrader/api/singular_noun` for single objects and `/formgrader/api/singular_noun/.../verb` for actions. I suppose 277 is a little inconsistent in that maybe it would make more sense to be `/formgrader/api/student_notebook_submisions/([^/]+)"? I'm afraid I don't see how 273 breaks the convention, though.